packages/devtools_app/lib/src/screens/dtd/services.dart

```diff
@@ -0,0 +1,305 @@
+}
+
+class _ManuallyCallServiceState extends State<_ManuallyCallService> {
+  final serviceController = TextEditingController();
+  final methodController = TextEditingController();
+  final paramsController = TextEditingController();
+
+  Map<String, Object?>? callResult;
+
+  @override
+  void initState() {
+    super.initState();
+    _maybePopulateSelectedService();
+  }
+
+  @override
+  void didUpdateWidget(covariant _ManuallyCallService oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.serviceMethod != widget.serviceMethod) {
+      callResult = null;
+      _maybePopulateSelectedService();
+    }
+  }
+
+  @override
+  void dispose() {
+    serviceController.dispose();
+    methodController.dispose();
+    paramsController.dispose();
+    super.dispose();
+  }
+
+  void _maybePopulateSelectedService() {
+    if (widget.serviceMethod != null) {
+      serviceController.text = widget.serviceMethod!.service ?? '';
+      methodController.text = widget.serviceMethod!.method;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(denseSpacing),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          _buildTextField('Service', serviceController, 'service_name'),
+          const SizedBox(height: denseSpacing),
+          _buildTextField('Method', methodController, 'method_name'),
+          const SizedBox(height: denseSpacing),
+          Row(
+            children: [
+              const Text('Additional params (json encoded):'),
+              const SizedBox(width: defaultSpacing),
+              Expanded(
+                child: DevToolsClearableTextField(
+                  controller: paramsController,
+                  hintText: '{"foo":"bar"}',
+                ),
+              ),
+            ],
+          ),
+          const SizedBox(height: defaultSpacing),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.end,
+            children: [
+              DevToolsButton(
+                label: 'Clear',
+                onPressed: () {
+                  setState(() {
+                    callResult = null;
+                    serviceController.clear();
+                    methodController.clear();
+                    paramsController.clear();
+                  });
+                },
+              ),
+              const SizedBox(width: denseSpacing),
+              DevToolsButton(
+                elevated: true,
+                label: 'Call Service',
+                onPressed: _callService,
+              ),
+            ],
+          ),
+          const PaddedDivider.thin(),
+          Expanded(
+            child: SingleChildScrollView(
+              child: Text(
+                callResult == null
+                    ? 'Call the service to view the response'
+                    : (callResult ?? {}).toString(),
```

Comment:
`callResult ?? {}` will never result in `{}` being returned. This can just be `callResult.toString()`.